### ISSUE TITLE
Threaded Server

### ISSUE THREAD

**Issue by cpcloud** [ContinuumIO/blaze#1149]
It'd be nice if the server 1) didn't block and 2) was able to concurrently handle more than a single connection (that doesn't necessarily mean that the underlying resource can handle concurrent reads, but many of them can).

There was a PR a while ago by @mrocklin and @llllllllll has a method for doing this via gunicorn #835

**Comment by llllllllll** [comment]
Just to note, the gunicorn method works by creating synchronous flask worker processes to handle each request; however, each individual worker can only handle requests serially.

**Comment by hhuuggoo** [comment]
gunicorn has threaded, workers as well, and async via gevent or tornado

**Comment by llllllllll** [comment]
Yup; I just wanted to note that the example linked was not really mutli-threaded, more so mutli-processed.
